### PR TITLE
feat(cli): filter vulnerability assessments by severity

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Add --severity flag to filter vulnerability assessment output

Usage:
`lacework vulnerability container show-assessment <container>  --severity low`

Closes https://github.com/lacework/go-sdk/issues/245

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: cli/cmd/vuln_container.go [hunk 1]
```diff
@@ -442,20 +448,26 @@ func buildVulnerabilityReportTable(assessment *api.VulnContainerAssessment) stri
 		),
 	)
 
-	if vulCmdState.Details || vulCmdState.Fixable || vulCmdState.Packages {
+	if vulCmdState.Details || vulCmdState.Fixable || vulCmdState.Packages || vulFiltersEnabled() {
```
Comment (afiune) [inline cli/cmd/vuln_container.go:451]: I think we have so many filters now that we could do what you did for compliance filters, abstract them into a function.

File: cli/cmd/vuln_container.go [hunk 2]
```diff
@@ -555,11 +593,15 @@ func vulContainerImageLayersToTable(image *api.VulnContainerImage) [][]string {
 		}
 	}
 
+	if vulFiltersEnabled() {
+		filteredOutput = fmt.Sprintf("%v of %v vulnerabilities showing \n", len(out), vulnsCount)
```
Comment (afiune) [inline cli/cmd/vuln_container.go:597]: I really like this footer, I kind of want it in the events list command. 😅

File: cli/cmd/vuln_container.go [hunk 3]
```diff
@@ -520,26 +544,40 @@ func vulContainerImagePackagesToTable(image *api.VulnContainerImage) [][]string
 		}
 	}
 
+	if vulFiltersEnabled() {
+		filteredOutput = fmt.Sprintf("%v of %v packages showing \n", len(out), packagesCount)
```
Comment (afiune) [inline cli/cmd/vuln_container.go:548]: I think the count of packages is wrong. use this command inside our tech-ally account:
```
lacework vuln ctr show sha256:4a69362cb4a07631be9d5688e81befa5a39e02d117411242c83617238393565e --severity high --packages
```
You will see:
```
  CVE COUNT   SEVERITY   PACKAGE   CURRENT VERSION   FIX VERSION
------------+----------+---------+-----------------+--------------
  2           High       curl      7.69.1-r1         7.69.1-r3
  1           High       curl      7.69.1-r1         7.69.1-r2
2 of 1 packages showing
```

**Review by afiune — COMMENTED** [review]
Pretty good! One issue with the packages footer and a couple nitpicks.

**Comment by afiune** [comment]
I fixed a test in `main`, I will rebase your PR.

File: cli/cmd/vuln_container.go (hunk 1 above)
Comment (dmurray-lacework) [reply cli/cmd/vuln_container.go:451]: Here, i've distinguished filters from the other flags details and packages in vulFiltersEnabled(). I could see fixable being pulled into the vulFiltersEnabled but I don't think details and packages quite fit.

Comment (dmurray-lacework) [reply cli/cmd/vuln_container.go]: Done for this fixible. I don't think packages and details should be pulled in.

File: cli/cmd/vuln_container.go [hunk 4]
```diff
@@ -477,69 +492,107 @@ func buildVulnerabilityReportTable(assessment *api.VulnContainerAssessment) stri
...
 }
 
-func vulContainerImagePackagesToTable(image *api.VulnContainerImage) [][]string {
+type packageTable struct {
+	cveCount       int
+	severity       string
+	packageName    string
+	currentVersion string
+	fixVersion     string
+}
```
Comment (afiune) [inline cli/cmd/vuln_container.go:501]: Nice!

**Review by afiune — APPROVED** [review]
tenor-71363449

**Pull request merged**